src/Analysis/Ast/Impl/Indexing/DirectoryFileReader.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+using Microsoft.Python.Core;
+using Microsoft.Python.Core.IO;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Python.Analysis.Indexing {
+    internal class DirectoryFileReader {
```

Comment:
This looks like fairly generic class useful in other cases. I think we should move it to `Python.Core.IO' and simply add method to `IDirectoryInfo`, such as 
```cs
IEnumerable<IFileSystemInfo> EnumerateFileSystemInfos(string[] includeFiles, string[] excludeFiles);
```